src/TestInfrastructure/Framework/TestLogger.cs

```diff
@@ -46,6 +46,17 @@ public void AssertPartialOutputStrings(params string[] orderedPartialOutputMessa
                 actualValue.Contains(expectedValue));
         }
 
+        public void AssertOutputStringExists(string expected)
+        {
+            this.outputStrings.Should().Contain(expected + Environment.NewLine); // All messages are postfixed by a newline
+        }
+
+        public void AssertPartialOutputStringExists(params string[] expected)
+        {
+            this.outputStrings.Should()
+                .Contain(msg => expected.All(partial => msg.Contains(partial)));
+        }
+
```

Comment:
Additional methods to help check for the existence of a specific full or partial string. The existing helper methods check the complete set of logged messages, which is quite fragile.